internal/tiltfile/k8s.go

```diff
@@ -274,7 +274,50 @@ func (s *tiltfileState) k8sResource(thread *starlark.Thread, fn *starlark.Builti
 	}
 }
 
+func (s *tiltfileState) isProbablyK8SResourceV1Call(args starlark.Tuple, kwargs []starlark.Tuple) (bool, string) {
+	var k8sResourceV1OnlyNames = map[string]bool{
+		"name":  true,
+		"yaml":  true,
+		"image": true,
+	}
+	for _, item := range kwargs {
+		name := string(item[0].(starlark.String))
+		if _, ok := k8sResourceV1OnlyNames[name]; ok {
+			return true, fmt.Sprintf("it was called with kwarg %q, which no longer exists", name)
+		}
+	}
+
+	// check positional args
+	// check if the second arg is yaml (v1) instead of a resource name (v2)
+	if args.Len() >= 2 {
+		switch x := args[1].(type) {
+		case starlark.Sequence:
+			return true, "second arg was a sequence"
+		case *blob:
+			return true, "second arg was a blob"
+		case starlark.String:
+			bs, err := ioutil.ReadFile(s.localPathFromString(string(x)).path)
+			if err == nil {
+				if bytes.ContainsRune(bs, '\n') {
+					return true, "second arg was a file containing a newline, which might be yaml, and can't be a valid resource name"
```

Comment:
...wait what is this case?

Reply:
added comments to try to answer this question